private/bufpkg/bufapimodule/module_resolver_test.go

```diff
@@ -0,0 +1,147 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bufapimodule
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/bufbuild/buf/private/bufpkg/bufmodule/bufmoduleref"
+	"github.com/bufbuild/buf/private/gen/proto/connect/buf/alpha/registry/v1alpha1/registryv1alpha1connect"
+	registryv1alpha1 "github.com/bufbuild/buf/private/gen/proto/go/buf/alpha/registry/v1alpha1"
+	"github.com/bufbuild/connect-go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type mockCommitServiceClient struct {
+	t       *testing.T
+	refResp *registryv1alpha1.GetRepositoryCommitByReferenceResponse
+}
+
+func (m *mockCommitServiceClient) ListRepositoryCommitsByBranch(
+	_ context.Context,
+	_ *connect.Request[registryv1alpha1.ListRepositoryCommitsByBranchRequest],
+) (*connect.Response[registryv1alpha1.ListRepositoryCommitsByBranchResponse], error) {
+	m.t.Error("unexpected call: ListRepositoryCommitsByBranch")
+	return nil, errors.New("unexpected call")
+}
+
+func (m *mockCommitServiceClient) ListRepositoryCommitsByReference(
+	_ context.Context,
+	_ *connect.Request[registryv1alpha1.ListRepositoryCommitsByReferenceRequest],
+) (*connect.Response[registryv1alpha1.ListRepositoryCommitsByReferenceResponse], error) {
+	m.t.Error("unexpected call: ListRepositoryCommitsByReference")
+	return nil, errors.New("unexpected call")
+}
+
+func (m *mockCommitServiceClient) GetRepositoryCommitByReference(
+	_ context.Context,
+	_ *connect.Request[registryv1alpha1.GetRepositoryCommitByReferenceRequest],
+) (*connect.Response[registryv1alpha1.GetRepositoryCommitByReferenceResponse], error) {
+	return connect.NewResponse(m.refResp), nil
+}
+
+func (m *mockCommitServiceClient) GetRepositoryCommitBySequenceId(
+	_ context.Context,
+	_ *connect.Request[registryv1alpha1.GetRepositoryCommitBySequenceIdRequest],
+) (*connect.Response[registryv1alpha1.GetRepositoryCommitBySequenceIdResponse], error) {
+	m.t.Error("unexpected call: GetRepositoryCommitBySequenceId")
+	return nil, errors.New("unexpected call")
+}
+
+func (m *mockCommitServiceClient) ListRepositoryDraftCommits(
+	_ context.Context,
+	_ *connect.Request[registryv1alpha1.ListRepositoryDraftCommitsRequest],
+) (*connect.Response[registryv1alpha1.ListRepositoryDraftCommitsResponse], error) {
+	m.t.Error("unexpected call: ListRepositoryDraftCommits")
+	return nil, errors.New("unexpected call")
+}
+
+func (m *mockCommitServiceClient) DeleteRepositoryDraftCommit(
+	_ context.Context,
+	_ *connect.Request[registryv1alpha1.DeleteRepositoryDraftCommitRequest],
+) (*connect.Response[registryv1alpha1.DeleteRepositoryDraftCommitResponse], error) {
+	m.t.Error("unexpected call: DeleteRepositoryDraftCommit")
+	return nil, errors.New("unexpected call")
+}
```

Comment:
```suggestion
type mockCommitServiceClient struct {
	registryv1alpha1connect.UnimplementedRepositoryCommitServiceHandler
	t       *testing.T
	refResp *registryv1alpha1.GetRepositoryCommitByReferenceResponse
}
```

Reply:
Don't I wish it were that easy. All methods are required to fulfill `registryv1alpha1connect.RepositoryCommitServiceClient`.

Reply:
But embedding `registryv1alpha1connect.UnimplementedRepositoryCommitServiceHandler` should give you at least stub methods for the rest, right? Am I misunderstanding?

Reply:
Yeah, I always put a space between fields and any embedded types, so I didn't see the suggestion's embed.